src/global/ObjectTest.java

```diff
@@ -0,0 +1,485 @@
+        return callMethod(methodName, arguments, modifiers, object, methodTestSentence, "");
+    }
+
+    public Object callMethod(String methodName, Object[][] arguments, Object object, Clause[] methodTestSentence) throws Throwable {
+        return callMethod(methodName, arguments, null, object, methodTestSentence, "");
+    }
+
+    public Object callMethod(String methodName, Object[][] arguments, String[] modifiers, Clause[] methodTestSentence, String incorrectMethodStructureErrorMessage) throws Throwable {
+        return callMethod(methodName, arguments, modifiers, null, methodTestSentence, incorrectMethodStructureErrorMessage);
+    }
+
+    public Object callMethod(String methodName, Object[][] arguments, Clause[] methodTestSentence, String incorrectMethodStructureErrorMessage) throws Throwable {
+        return callMethod(methodName, arguments, null, null, methodTestSentence, incorrectMethodStructureErrorMessage);
+    }
+
+    public Object callMethod(String methodName, String[] modifiers, Object object, Clause[] methodTestSentence, String incorrectMethodStructureErrorMessage) throws Throwable {
+        return callMethod(methodName, null, modifiers, object, methodTestSentence, incorrectMethodStructureErrorMessage);
+    }
+
+    public Object callMethod(String methodName, Object object, Clause[] methodTestSentence, String incorrectMethodStructureErrorMessage) throws Throwable {
+        return callMethod(methodName, null, null, object, methodTestSentence, incorrectMethodStructureErrorMessage);
+    }
+
+    public Object callMethod(String methodName, Object[][] arguments, String[] modifiers, Object object,
+                             Clause[] methodTestSentence, String incorrectMethodStructureErrorMessage) throws Throwable {
+        Class<?>[] argsClass = getArgumentClasses(arguments);
+        Object[] args = getArguments(arguments);
+        try {
+            Method objectMethodInvoke = objectClass.getMethod(methodName, argsClass);
+            if (!Objects.isNull(modifiers)) {
+                for (String modifier : modifiers) {
+                    assertTrue(hasModifier(objectMethodInvoke, modifier),
+                            String.join(" ", "Your", methodName, "method in the", objectClass.getSimpleName(),
+                                    "class does not have the correct modifiers."));
+                }
+            }
+            ByteArrayOutputStream methodOutput = new ByteArrayOutputStream();
+            System.setOut(new PrintStream(methodOutput));
+            Object output = objectMethodInvoke.invoke(object, args);
+            methodOutputFollowsCorrectStructure(methodName, methodTestSentence, incorrectMethodStructureErrorMessage, methodOutput);
+            return output;
+        } catch (NoSuchMethodException e) {
+            fail(String.join("", "Your ", objectClass.getSimpleName(),
```

Comment:
Why do we have both `fail` and `_fail`? Whats the difference?

Reply:
_fail allows us to use two messages. The first is the error message the user will see, and the second logs a developer message to the console which tells them if they made a mistake with something, or gives them more description of the error.